pandas/tests/indexes/timedeltas/test_setops.py

```diff
@@ -73,3 +76,92 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         expected = timedelta_range('1 day 01:00:00', periods=3, freq='h')
         tm.assert_index_equal(result, expected)
+
+    @pytest.mark.parametrize("sort", [None, False])
+    def test_intersection_equal(self, sort):
+        # GH 24471 Test intersection outcome given the sort keyword
+        # for equal indicies intersection should return the original index
+        first = timedelta_range('1 day', periods=4, freq='h')
+        second = timedelta_range('1 day', periods=4, freq='h')
+        intersect = first.intersection(second, sort=sort)
+        if sort is None:
+            tm.assert_index_equal(intersect, second.sort_values())
+        assert tm.equalContents(intersect, second)
+
+        # Corner cases
+        inter = first.intersection(first, sort=sort)
+        assert inter is first
+
+    @pytest.mark.parametrize("period_1, period_2", [(0, 4), (4, 0)])
+    @pytest.mark.parametrize("sort", [None, False])
+    def test_intersection_zero_length(self, period_1, period_2, sort):
+        # GH 24471 test for non overlap the intersection should be zero length
+        index_1 = timedelta_range('1 day', periods=period_1, freq='h')
+        index_2 = timedelta_range('1 day', periods=period_2, freq='h')
+        inter = index_1.intersection(index_2, sort=sort)
+        tm.assert_index_equal(timedelta_range('1 day', periods=0, freq='h'),
+                              inter)
+
+    @pytest.mark.parametrize("rng, expected",
+                             # if target has the same name, it is preserved
+                             [(timedelta_range('1 day', periods=5,
+                                               freq='h', name='idx'),
+                               timedelta_range('1 day', periods=4,
+                                               freq='h', name='idx')),
+                              # if target name is different, it will be reset
+                              (timedelta_range('1 day', periods=5,
+                                               freq='h', name='other'),
+                               timedelta_range('1 day', periods=4,
+                                               freq='h', name=None)),
+                              # if no overlap exists return empty index
+                              (timedelta_range('1 day', periods=10,
+                                               freq='h', name='idx')[5:],
+                               TimedeltaIndex([], name='idx'))
+                              ])
+    @pytest.mark.parametrize("sort", [None, False])
```

Comment:
do we not have generic test_setops (that hit Timedeltas)?

Reply:
@jreback Again, not 100% clear where to find that generic test_setops.

Reply:
pandas/tests/indexes/test_setops.py

Reply:
No there exists no pandas/tests/indexes/test_setops.py , I checked this before but then decided to add the tests here. Happy to move the test_setops that is in pandas/tests/indexes/timedeltas one level up but was not sure that this is appropriate.

Reply:
sorry these are in: pandas/tests/indexes/test_base.py

I think we need to move them to a new file though. If you can do it here would be great, otherwise need to consolidate all of this as a followup

Reply:
I am happy to work on it but maybe as a followup to this issue if that is ok?